### PR TITLE
Add labeling v1alpha3

### PR DESCRIPTION
**Pull request by anvddriesch**
**What type of PR is this?**

**What this PR does / why we need it**:

This is WIP. We want to push the images to our giantswarm quay registry and add the label filtering.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

File: controllers/azurejson_machinepool_controller.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ type AzureJSONMachinePoolReconciler struct {
 	Log              logr.Logger
 	Recorder         record.EventRecorder
 	ReconcileTimeout time.Duration
+	WatchFilterValue string
```
Comment (MarcelMue) [inline controllers/azurejson_machinepool_controller.go:51]: Needs `WithEventFilter(predicates.ResourceHasFilterLabel(r.Log, r.WatchFilterValue)).` in line 57

File: controllers/azurejson_machinetemplate_controller.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ type AzureJSONTemplateReconciler struct {
 	Log              logr.Logger
 	Recorder         record.EventRecorder
 	ReconcileTimeout time.Duration
+	WatchFilterValue string
```
Comment (MarcelMue) [inline controllers/azurejson_machinetemplate_controller.go:51]: Needs `WithEventFilter(predicates.ResourceHasFilterLabel(r.Log, r.WatchFilterValue)).` in line 58

**Review by MarcelMue — APPROVED** [review]
Let's try and test this I guess?

**Comment by anvddriesch** [comment]
Let's do that. I'll see if i can add it to the app.

**Review by fiunchinho — APPROVED** [review]

**Comment by fiunchinho** [comment]
@anvddriesch now that the Hive Sprint is over, should we take over this PR in Celestial?

**Comment by MarcelMue** [comment]
> @anvddriesch now that the Hive Sprint is over, should we take over this PR in Celestial?

Antonia is on vacation this week, but yes - do it :rocket:

**Comment by tuommaki** [comment]
@MarcelMue Do you know if there's anything missing from here?

**Comment by MarcelMue** [comment]
This should be working as intended right now. The PR isn't merged to make updating with upstream easier (so we can get new commits in master and then merge into this branch)

**Comment by nprokopic** [comment]
> This should be working as intended right now. The PR isn't merged to make updating with upstream easier (so we can get new commits in master and then merge into this branch)

Currently `release-0.4` branch in our fork matches the upstream release v0.4.13, so it's almost ideal point for merging this PR, because then we get "clean" upstream v0.4.13 + watch label filtering.

Then we can continue syncing our releases with upstream and have e.g. our v0.4.14 to match upstream v0.4.14 etc.

**Comment by anvddriesch** [comment]
:) If it's okay I'll leave it to celestials judgement to merge this PR.

**Pull request merged**